### PR TITLE
103 refactoring menu and data fetching

### PR DESCRIPTION
**Pull request by kardwen**
Refactor menu and display message if request fails

- Rename variables and functions so that names are specific
- Send request string to the MainPage and fetch data there
- Fix a bug that created faulty requests
- Add warning for when the server is not responding

 Replace props mutating slider handling and restructure code

 Change configuration descriptions

- change smooting parameter again

### CODE REVIEW

File: frontend/src/components/MainPage.vue [hunk 1]
```diff
@@ -84,7 +92,7 @@ export default {
           active: true, // wether the layer is currently selected by the user
           // the text that shall be displayed when the user hovers over the info button
           infoLabel:
-            "Move the slider to remove all areas <br/>that have a certain <b>distance to museums</b>.",
+            "Move the slider to remove all Areas <br/>that have a certain <b>distance to museums</b>.",
```
Comment (jbalzer12) [inline frontend/src/components/MainPage.vue:95]: areas has to be written small as a noun

File: frontend/src/components/MainPage.vue [hunk 2]
```diff
@@ -95,13 +103,11 @@ export default {
           band: 1,
           active: true,
           infoLabel:
-            "Move the slider to remove all areas <br/>that have a certain <b>distance to theaters</b>.",
+            "Move the slider to remove all Areas <br/>that have a certain <b>distance to theaters</b>.",
```
Comment (jbalzer12) [inline frontend/src/components/MainPage.vue:106]: same as above

File: frontend/src/components/MenuView.vue [hunk 3]
```diff
@@ -10,24 +10,32 @@
...
+    <v-alert
+      v-if="resultAreasRequestFailed"
+      text
+      color="rgb(25,118,210)"
+      colored-border
+      type="info"
+      class="py-2 px-2 mb-0 mt-3"
+    >
+      Request to the server failed.
+    </v-alert>
```
Comment (jbalzer12) [inline frontend/src/components/MenuView.vue:38]: how to provoke this? maybe we should do a warning or error here and not only an info. What do you think?

Comment (kardwen) [reply frontend/src/components/MenuView.vue:38]: Shut down the docker container

File: frontend/src/components/MainPage.vue (hunk 1 above)
Comment (kardwen) [reply frontend/src/components/MainPage.vue:95]: changed on accident

**Comment by kardwen** [comment]
The changes are pretty important as well as the fix for the bus station issue, that's why I merged it now.

File: frontend/src/components/MenuView.vue (hunk 3 above)
Comment (jbalzer12) [reply frontend/src/components/MenuView.vue:38]: I tried that but it didn’t work. Maybe I made something wrong. I check tomorrow

**Comment by jbalzer12** [comment]
> The changes are pretty important as well as the fix for the bus station issue, that's why I merged it now.

Please wait the next time until you got an approve. And what about the bus station issue?

**Comment by kardwen** [comment]
> > The changes are pretty important as well as the fix for the bus station issue, that's why I merged it now.
> 
> Please wait the next time until you got an approve. And what about the bus station issue?

Everything is documented in my commit messages, the branch that you merged earlier today contained structural problems which led to an error that the state of the two buttons for changing the layer visibility diverged at some point.

In this merge I fixed older problems that have the same root problems: redundancy in handling the state and properties. For example, eslint was disabled in MenuView to circumvent issues that arised from this. This led to a code structure that was hard to read and hard to debug.

Therefore merging couldn't wait and additionally this request was open for some time. We should discuss code quality later in more detail because it is even more important than just writing comments.

**Comment by jbalzer12** [comment]
> > > The changes are pretty important as well as the fix for the bus station issue, that's why I merged it now.
> > 
> > 
> > Please wait the next time until you got an approve. And what about the bus station issue?
> 
> Everything is documented in my commit messages, the branch that you merged earlier today contained structural problems which led to an error that the state of the two buttons for changing the layer visibility diverged at some point.
 
That's the reason I opened a pull request and had it open for multiple days. Next time I would prefer you make me pay attention to errors in my code so I can improve my style and learn from it. 

> In this merge I fixed older problems that have the same root problems: redundancy in handling the state and properties. For example, eslint was disabled in MenuView to circumvent issues that arised from this. This led to a code structure that was hard to read and hard to debug.
> 
> Therefore merging couldn't wait and additionally this request was open for some time. We should discuss code quality later in more detail because it is even more important than just writing comments.

The pull requests wasn't that long open and especially it was weekend in the holidays. So these are conditions under which it may takes some more time to react.

**Pull request merged**